eclipse-che/che-operator
# feat: Make DevWorkspaceRouting a pre-requisite of che-operator.

### What does this PR do?
In order to be able to merge codebases of DWCO and CO, we need to pre-install the devworkspacerouting CRD into the cluster, if supported by the target platform (i.e. this should NOT be done on OpenShift 3).

This PR includes the the DevWorkspaceRouting CRD in the deployment files of the che-operator and updates (or at least tries to) the OLM deployment scripts to include it there, too.

I am personally unfamiliar with OLM so I would appreciate any and all help with testing this PR.

Note that currently, the DevWorkspaceRouting CRD is just contained in the repo. Alternatively, we could think about downloading it from DWO, but then we should be very careful about downloading the correct version (should we use the latest version during or should we stick to some released version?).

### Screenshot/screencast of this PR
N/A


### What issues does this PR fix or reference?
eclipse/che#20088

### How to test this PR?
Run `./deploy.sh`. The cluster should contain the `devworkspaceroutings.controller.devfile.io` CRD after that.

### PR Checklist
          
[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)


### Reviewers

Reviewers, please comment how you tested the PR when approving it.
